File: infra/src/main/java/com/beat/infra/persistence/promotion/repository/PromotionJpaRepository.java
package com.beat.infra.persistence.promotion.repository;

import java.util.List;
import java.util.Optional;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Modifying;
import org.springframework.data.jpa.repository.Query;
import org.springframework.data.repository.query.Param;
import org.springframework.transaction.annotation.Transactional;

import com.beat.domain.promotion.domain.CarouselNumber;
import com.beat.infra.persistence.promotion.entity.PromotionJpaEntity;

public interface PromotionJpaRepository extends JpaRepository<PromotionJpaEntity, Long> {

	@Modifying(clearAutomatically = true)
	@Transactional
	@Query("DELETE FROM Promotion p WHERE p.id IN :promotionIds")
	void deleteByPromotionIds(@Param("promotionIds") List<Long> promotionIds);

	@Modifying(clearAutomatically = true)
	@Transactional
	@Query("DELETE FROM Promotion p WHERE p.performanceId = :performanceId")
	void deleteByPerformanceId(@Param("performanceId") Long performanceId);

	@Query("SELECT p FROM Promotion p WHERE p.carouselNumber = :carouselNumber")
	Optional<PromotionJpaEntity> findByCarouselNumber(@Param("carouselNumber") CarouselNumber carouselNumber);
}

File: core/infra/src/main/kotlin/com/beat/infra/persistence/promotion/repository/PromotionJpaRepository.kt
package com.beat.infra.persistence.promotion.repository

import com.beat.domain.promotion.model.CarouselNumber
import com.beat.infra.persistence.promotion.entity.PromotionJpaEntity
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.Modifying
import org.springframework.data.jpa.repository.Query
import org.springframework.data.repository.query.Param
import org.springframework.transaction.annotation.Transactional
import java.util.Optional

interface PromotionJpaRepository : JpaRepository<PromotionJpaEntity, Long> {

    @Modifying(clearAutomatically = true)
    @Transactional
    @Query("DELETE FROM Promotion p WHERE p.id IN :promotionIds")
    fun deleteByPromotionIds(@Param("promotionIds") promotionIds: List<Long>)

    @Modifying(clearAutomatically = true)
    @Transactional
    @Query("DELETE FROM Promotion p WHERE p.performanceId = :performanceId")
    fun deleteByPerformanceId(@Param("performanceId") performanceId: Long?)

    @Query("SELECT p FROM Promotion p WHERE p.carouselNumber = :carouselNumber")
    fun findByCarouselNumber(
        @Param("carouselNumber") carouselNumber: CarouselNumber,
    ): Optional<PromotionJpaEntity>
}
